list-agreements.lua

```diff
@@ -41,19 +40,50 @@ function get_petition_date(agr)
     julian_day = math.floor(agr_year_tick / 1200) + 1
     agr_month = math.floor(julian_day / 28) + 1
     agr_day = julian_day % 28
-    return agr_year.."-"..agr_month.."-"..agr_day
+    return string.format("%03d-%02d-%02d",agr_year, agr_month, agr_day)
+end
+
+function get_petition_age(agr)
+    agr_year_tick = agr.details[0].year_tick
+    agr_year = agr.details[0].year
+    cur_year_tick = df.global.cur_year_tick
+    cur_year = df.global.cur_year
+    --delta, check to prevent off by 1 error, not validated
+    if cur_year_tick > agr_year_tick then
+        del_year = cur_year - agr_year
+        del_year_tick = cur_year_tick - agr_year_tick
+    else
+        del_year = cur_year - agr_year - 1
```

Comment:
That's a fair point; i was trying to avoid -ve month/day values in the delta

My understanding is that `_year_tick` resets on year rollover (otherwise it would be more suitably named, such as `world_tick`) and thus if the day-of-year of the current year is before the yearly anniversary of the agreement, the delta would be negative.

However, this is of course mooted if `_year_tick` does not reset

Reply:
ah, ok, that makes sense. Yes, I was looking at this as if `_year_tick` did not reset, but I think you're right that it does.